packages/api/lambdas/process-s3-dead-letter-archive.js

```diff
@@ -31,28 +31,28 @@ const generateNewArchiveKeyForFailedMessage = (failedKey) => {
  * Transfers unprocessed dead letters in the bucket to new location
  * and deletes dead letters from old archive path
  *
- * @param {string}   [deadLetterMessage] - unprocessed dead letter message
+ * @param {string}   [deadLetterObject] - unprocessed dead letter object
  * @param {string}   [bucket] - S3 bucket
  * @returns {Promise<void>}
  */
-const transferUnprocessedMessage = async (deadLetterMessage, bucket) => {
+const transferUnprocessedMessage = async (deadLetterObject, bucket) => {
   // Save allFailedKeys messages to different location
-  const s3KeyForFailedMessage = generateNewArchiveKeyForFailedMessage(deadLetterMessage.Key);
+  const s3KeyForFailedMessage = generateNewArchiveKeyForFailedMessage(deadLetterObject.Key);
   try {
     log.info(`Attempting to save messages that failed to process to ${bucket}/${s3KeyForFailedMessage}`);
-    await S3.s3PutObject({
+    await S3.s3CopyObject({
       Bucket: bucket,
       Key: s3KeyForFailedMessage,
-      Body: deadLetterMessage.Body,
+      CopySource: `${bucket}/${deadLetterObject.Key}`,
```

Comment:
> deadLetterObject would be a failed execution or something

Yeah, exactly! (I elaborate more below)

>  and its basically being transferred in the first place to a new bucket in order to isolate it from the non-unprocessed dead letters.

Yep, because having to sift through the already processed dead letters was getting unwieldy for some users

> I assume through your research or investigation, it was concluded that copy > put (in this context), but is it in a way in which put just straight up doesn't work anymore or more like copy is just better/more efficient

Yeah exactly, so I opted for copying the archived message from the previous location to the new location if it failed to process and it was in part due to the object format potentially being somewhat gnarly. For example, you can see here https://github.com/nasa/cumulus/blob/b58c0051f05a5234bd9cff93390ae54efdeddd24/packages/api/lambdas/process-s3-dead-letter-archive.js#L96 the `processDeadLetterArchive` function will call `unwrapDeadLetterCumulusMessage` which is required because the message _could_ be a CloudWatch event wrapped in an SQS Message https://github.com/nasa/cumulus/blob/master/packages/message/src/DeadLetterMessage.ts#L26 

@Nnaga1

Reply:
> Other than the stuff I asked, the changes themselves look solid, is there anything else needed to be done before approval?

Ad hoc testing I think would be a good thing to do as well @Nnaga1

Reply:
> thanks for the explanation! so basically put would mess with/change the dead letter object in some way copy doesn't, which would cause issues, that makes sense

So for more context, the previous bug existed because the shape of `deadLetterObject` actually didn't contain a `.Body` property, so it was literally `undefined` 🤦‍♀️ https://github.com/nasa/cumulus/blob/master/packages/api/lambdas/process-s3-dead-letter-archive.js#L46

 and yeah the different possible structures of the object would be more cumbersome to parse + put rather than just directly copy